internal/bminventory/inventory.go

```diff
@@ -1517,6 +1531,26 @@ func (b *bareMetalInventory) refreshClusterHosts(ctx context.Context, cluster *c
 	return nil
 }
 
+func (b *bareMetalInventory) refreshInventory(ctx context.Context, cluster *common.Cluster, host *models.Host, db *gorm.DB) error {
+	log := logutil.FromContext(ctx, b.log)
+	if host.Inventory != "" {
+		err := b.hostApi.RefreshInventory(ctx, cluster, host, db)
+		if err != nil {
+			log.WithError(err).Errorf("failed to update inventory of host %s cluster %s", host.ID, cluster.ID.String())
+			switch err := err.(type) {
+			case *common.ApiErrorResponse:
+				if err.StatusCode() != http.StatusConflict {
+					return err
+				}
+				log.Infof("ignoring wrong status error (%v) for host %s in cluster %s", err, host.ID, cluster.ID.String())
```

Comment:
In the code common to `RefreshInventory` and `UpdateInventory` there is a precondition on host's status that on failure returns `StatusConflict`. In case of cluster update we don't want to fail the whole update if host can't be (according to business rules) updated - an example of that is disabled host.

Reply:
Could you please add it as a comment?